src/Raven.Server/Documents/Queries/Dynamic/CollectionQueryRunner.cs

```diff
@@ -136,6 +136,10 @@ private async ValueTask ExecuteCollectionQueryAsync(QueryResultServerSide<Docume
 
                 var fieldsToFetch = new FieldsToFetch(query, null, IndexType.None);
                 var includeDocumentsCommand  = new IncludeDocumentsCommand(Database.DocumentsStorage, context.Documents, query.Metadata.Includes, fieldsToFetch.IsProjection);
+                
+                if (resultToFill is StreamQueryResult<Document> && resultToFill.IncludedPaths != null && resultToFill.IncludedPaths.Length > 0)
```

Comment:
I think I would prefer to introduce inside QueryResultServerSide some kind of a property e.g. IsStream maybe?

Reply:
Then we could use it in other query runners where we already throw this exception

Reply:
I moved it to `StreamingHandler`, so we throw exception immediately after parsing query